### PR TITLE
Bootstrap

### PR DESCRIPTION
**Pull request by vupil**
Added examples and styled the file using styler

### CODE REVIEW

File: R/bootstrap.R [hunk 1]
```diff
@@ -0,0 +1,647 @@
...
+      )
+    )
+  }
+  else{
+    cli::cli_alert_danger(
+      cli::col_red(
+        "all models not saved. resume bootstrapping using saved models at {paste0(getwd(),'/', fnameModelsEnsemble)}"
+      )
+    )
+    stop('aborting...')
```
Comment (mattfidler) [inline R/bootstrap.R]: Tell why it it was aborted, and use `call.=FALSE`

File: R/bootstrap.R [hunk 2]
```diff
@@ -0,0 +1,647 @@
...
+#'
+#' @return returns a string representing the method used by nlmixr for fitting the given model
+#'
+#' @author Vipul Mann, Matthew Fidler
+#'
+#' @examples
+#' getFitMethod(fit)
+#'
+getFitMethod = function(fit) {
+  methodsList = c('nlmixrFOCEi', 'nlmixrNlmeUI', 'nlmixrSaem')
```
Comment (mattfidler) [inline R/bootstrap.R]: These need to be be added:
| method | map |
|------------|------|
| `'nlmixrFOCE'` | `'foce'` | 
| `'nlmixrFOi` | `'foi'` |
| `'nlmixrFO'` | `'fo'` |
| `'nlmixrPosthoc'` | `'posthoc'` |

**Comment by mattfidler** [comment]
The tests we went over are missing in this pull request.

**Comment by mattfidler** [comment]
Use the User's original control settings to do this with `fit` object you would do:

```r
.ctl <- fit$origControl
.ctl$print <- 0
.ctl$covMethod <- ""
.ctl$calcTables <- FALSE
```

This way:
- printing is suppressed
- the covariance calculation is suppressed
- Extra table items are not calculated.

File: R/bootstrap.R [hunk 3]
```diff
@@ -0,0 +1,647 @@
...
+  }))
+}
+
+
+#' Fitting multiple bootstrapped models
+#'
+#' @param fit the nlmixr fit object
+#' @param numModels an integer giving the number of bootstrapped models to be fit; default value is 3
+#' @param nSampIndiv an integer specifying the number of samples in each bootstrapped sample; default is the number of unique subjects in the original dataset
+#' @param uidCol a string representing the unique ID of each subject in the data; default values is 'ID'
```
[the remark this answers is not in the document]
Comment (vupil) [reply R/bootstrap.R:170]: Removed this argument from both bootstrapFit and modelBootstrap functions

File: R/bootstrap.R [hunk 4]
```diff
@@ -0,0 +1,647 @@
...
+  })
+  
+  unlist(res)
+}
+
+#' Main bootstrap function that calls all the other dependent bootstrap functions to sample from a given dataset, fit
+#' multiple models, compute aggregate values, assign them to the nlmixr environment, and update the fit object.
+#'
+#' @param fit
+#' @param numModels an integer giving the number of bootstrapped models to be fit; default value is 3
```
[the remark this answers is not in the document]
Comment (vupil) [reply R/bootstrap.R:20]: Replaced ```numModels``` with ```nboot``` everywhere

File: R/bootstrap.R [hunk 5]
```diff
@@ -0,0 +1,647 @@
...
+  summaryList
+  
+}
+
+#' Print a well-formatted summary for the bootstrap models
+#'
+#' @param x the summary object returned by the getBootstrapSummary() function
+#'
+#' @author Vipul Mann, Matthew Fidler
+#'
```
[the remark this answers is not in the document]
Comment (vupil) [reply R/bootstrap.R:559]: changed the function name to ```print.nlmixrBootstrapSummary```. But the function doesn't return anything - it just prints the summary. How do we define the return class in that case? I have changed the class of ```x``` returned by ```getBootstrapSummary``` to "nlmixrBoostrapSummary". Is that okay?

**Comment by mattfidler** [comment]
Thanks @vupil 

Merged in the boostrapping.  I will test it some more too.

**Pull request merged**